src/generic/hooks/useIframeBehavior.tsx

```diff
@@ -1,13 +1,18 @@
-import { useCallback, useEffect } from 'react';
+import { useCallback, useEffect, useState } from 'react';
 import { logError } from '@edx/frontend-platform/logging';
-// eslint-disable-next-line import/no-extraneous-dependencies
-import { useKeyedState } from '@edx/react-unit-test-utils';
-
+import StrictDict from '@src/editors/utils/StrictDict';
 import { useLoadBearingHook } from './useLoadBearingHook';
-import { iframeStateKeys, iframeMessageTypes } from '../../constants';
+import { iframeMessageTypes } from '../../constants';
 import { UseIFrameBehaviorReturnTypes, UseIFrameBehaviorTypes } from '../types';
 import { useEventListener } from './useEventListener';
 
+export const ifameBehaviorState = StrictDict({
+    iframeHeight: (val) => useState<number>(val), // eslint-disable-line
+    hasLoaded: (val) => useState<boolean>(val), // eslint-disable-line
+    showError: (val) => useState<boolean>(val), // eslint-disable-line
+    windowTopOffset: (val) => useState<number | null>(val), // eslint-disable-line
+});
```

Comment:
I'm not a huge fan of this, but I know you're dealing with the existing structure and I agree this is still an improvement.

Please make these tweaks though:

```suggestion
/**
 * Internal state hooks for useIFrameBehavior.
 * Defined and exported here so they can be mocked/manipulated in tests.
 */
export const iframeBehaviorState = StrictDict({
  /* eslint-disable react-hooks/rules-of-hooks */
  iframeHeight: (val) => useState<number>(val),
  hasLoaded: (val) => useState<boolean>(val),
  showError: (val) => useState<boolean>(val),
  windowTopOffset: (val) => useState<number | null>(val),
  /* eslint-enable react-hooks/rules-of-hooks */
});
```

- spelling typo
- indentation fix
- explain why we're doing this hacky thing
- be explicit about which eslint rule we're suppressing